osquery/logger/plugins/aws_firehose.cpp

```diff
@@ -12,6 +12,12 @@
 
 #include <boost/algorithm/string/join.hpp>
 
+#ifdef WIN32
+// AWS SDK provides deprecation warnings at compile-time, and osquery
+// build treats warnings as errors, so have to turn these off.
+#define AWS_DISABLE_DEPRECATION
```

Comment:
I would prefer to add this to the `CMakeLists.txt` instead of defining before each include.